### ISSUE TITLE
KeyError when nonexistant runner function is called

### ISSUE THREAD

**Issue by terminalmage** [saltstack/salt#21460]
```
# salt-run jobs.print 20150309163238214802
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 'jobs.print'
Traceback (most recent call last):
  File "/home/erik/saltdev/bin/salt-run", line 7, in <module>
    execfile(__file__)
  File "/home/erik/git/salt/scripts/salt-run", line 10, in <module>
    salt_run()
  File "/home/erik/git/salt/salt/scripts.py", line 243, in salt_run
    client.run()
  File "/home/erik/git/salt/salt/cli/run.py", line 54, in run
    runner.run()
  File "/home/erik/git/salt/salt/runner.py", line 140, in run
    self.functions[low['fun']],
  File "/home/erik/git/salt/salt/loader.py", line 873, in __getitem__
    mod = super(LazyLoader, self).__getitem__(key)
  File "/home/erik/git/salt/salt/utils/lazy.py", line 75, in __getitem__
    raise KeyError(key)
KeyError: 'jobs.print'
Traceback (most recent call last):
  File "/home/erik/saltdev/bin/salt-run", line 7, in <module>
    execfile(__file__)
  File "/home/erik/git/salt/scripts/salt-run", line 10, in <module>
    salt_run()
  File "/home/erik/git/salt/salt/scripts.py", line 243, in salt_run
    client.run()
  File "/home/erik/git/salt/salt/cli/run.py", line 54, in run
    runner.run()
  File "/home/erik/git/salt/salt/runner.py", line 140, in run
    self.functions[low['fun']],
  File "/home/erik/git/salt/salt/loader.py", line 873, in __getitem__
    mod = super(LazyLoader, self).__getitem__(key)
  File "/home/erik/git/salt/salt/utils/lazy.py", line 75, in __getitem__
    raise KeyError(key)
KeyError: 'jobs.print'
```

**Comment by terminalmage** [comment]
This also applies to remote execution functions as well, looking into the LazyLoader code now.

**Comment by terminalmage** [comment]
Ignore that, I ran the wrong CLI command, this is limited to runner functions.

**Comment by terminalmage** [comment]
OK, I have a fix in progress for this, will have a pull request in this afternoon

**Comment by terminalmage** [comment]
Fix is in #21508.

How to test the fix:

```
root v(saltdev) ~                                                           [14:48:50]
(⌐■_■) # salt-run fileserver.envs backend=roots
- base
- dev
root v(saltdev) ~                                                           [14:49:12]
(⌐■_■) # salt-run fileserver.thisfuncdoesntexist backend=roots
'fileserver.thisfuncdoesntexist' is not available
```

**Comment by rallytime** [comment]
I have tested this fix and it is working. Thanks @terminalmage!
